File: src/api/events.ts
import { apiClient } from "./client"

export interface ChapterEvent {
  id: string
  title: string
  category: "Workshop" | "Hackathon" | "Symposium" | "Seminar" | "Conference" | "Webinar" | "Chapter Event" | "Other"
  date: string
  startTime?: string
  endTime?: string
  time?: string
  start_time?: string
  end_time?: string
  location: string
  mode: "In-Person" | "Online" | "Hybrid"
  description: string
  image?: string
  registerUrl?: string
  externalUrl?: string
  register_url?: string
  external_url?: string
  speaker?: string
  speakerRole?: string
  speaker_role?: string
  deadline?: string
  featured?: boolean
  status: "upcoming" | "completed"
  published?: boolean
  created_at?: string
  updated_at?: string
}

export const eventsApi = {
  getAll: () => apiClient<ChapterEvent[]>("/api/events"),
  getById: (id: string) => apiClient<ChapterEvent>(`/api/events/${id}`),
  create: (data: Partial<ChapterEvent>) =>
    apiClient<ChapterEvent>("/api/events", {
      method: "POST",
      body: JSON.stringify(data),
      requireAuth: true,
    }),
  update: (id: string, data: Partial<ChapterEvent>) =>
    apiClient<ChapterEvent>(`/api/events/${id}`, {
      method: "PUT",
      body: JSON.stringify(data),
      requireAuth: true,
    }),
  delete: (id: string) =>
    apiClient<{ message: string; id: string }>(`/api/events/${id}`, {
      method: "DELETE",
      requireAuth: true,
    }),
  duplicate: (id: string) =>
    apiClient<ChapterEvent>(`/api/events/${id}/duplicate`, {
      method: "POST",
      requireAuth: true,
    }),
}

File: backend/app/schemas/events.py
from typing import Optional
from pydantic import BaseModel, ConfigDict
import datetime


class EventBase(BaseModel):
    title: str
    category: str = "Workshop"
    date: str
    start_time: Optional[str] = None
    end_time: Optional[str] = None
    time: Optional[str] = None
    location: str = "SSN College of Engineering"
    mode: str = "In-Person"
    description: str
    image: Optional[str] = None
    register_url: Optional[str] = None
    external_url: Optional[str] = None
    speaker: Optional[str] = None
    speaker_role: Optional[str] = None
    deadline: Optional[str] = None
    featured: Optional[bool] = False
    status: Optional[str] = "upcoming"
    published: Optional[bool] = True


class EventCreate(EventBase):
    pass


class EventUpdate(BaseModel):
    title: Optional[str] = None
    category: Optional[str] = None
    date: Optional[str] = None
    start_time: Optional[str] = None
    end_time: Optional[str] = None
    time: Optional[str] = None
    location: Optional[str] = None
    mode: Optional[str] = None
    description: Optional[str] = None
    image: Optional[str] = None
    register_url: Optional[str] = None
    external_url: Optional[str] = None
    speaker: Optional[str] = None
    speaker_role: Optional[str] = None
    deadline: Optional[str] = None
    featured: Optional[bool] = None
    status: Optional[str] = None
    published: Optional[bool] = None


class EventResponse(EventBase):
    id: str
    created_at: Optional[datetime.datetime] = None
    updated_at: Optional[datetime.datetime] = None

    model_config = ConfigDict(from_attributes=True)
